### PR TITLE
Export pure() action creator

### PR DESCRIPTION
**Pull request by davidkpiano**

### CODE REVIEW

File: .changeset/old-coins-kneel.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'xstate': minor
+---
+
+Added support for the `pure()` action creator.
```
Comment (Andarist) [inline .changeset/old-coins-kneel.md]: Would be great to include a short description of what it is good for - it's really valuable piece of information for a changelog, shortens the learning loop (one doesnt have to jump into the docs just to get a general sense of what it is useful for).

**Comment by davidkpiano** [comment]
Yes, `pure()` should be `script()`, so you're right, we can probably hold off on this, but the other major use case is "send an event to all dynamically generated actors" which is best represented as:

```js
actions: pure((ctx, e) => {
  return ctx.exampleActors.map(actor => {
    return send(e, { to: actor })
  });
})
```

**Comment by Andarist** [comment]
@davidkpiano do we want to implement equivalent for SCXML's `<foreach/>`? the mentioned use case would be covered by such, so wondering if there are really any other use cases for `pure`? It's definitely easier to write actions using pure rather than with if+foreach combos, but other aspects can suffer from it, such as visualization

**Comment by davidkpiano** [comment]
> @davidkpiano do we want to implement equivalent for SCXML's <foreach/>? the mentioned use case would be covered by such, so wondering if there are really any other use cases for pure? It's definitely easier to write actions using pure rather than with if+foreach combos, but other aspects can suffer from it, such as visualization

Might be a good idea, but it's not that straightforward to implement.

There are many more use-cases for `pure()`, but of course the stricter actions should be preferred.

`pure()` actually allows for better visualization than having that dynamic logic in a random action (we can visualize what it would be in various states), but of course, it'd be better through stricter actions. I'm not too concerned about visualization.

**Comment by davidkpiano** [comment]
This PR is stale (`pure()` and docs already merged in)

**Pull request closed without merging**